### PR TITLE
8.4 modal window update student or teacher profile kozhukh

### PR DESCRIPTION
**Pull request by OksanaKozhukh**

### CODE REVIEW

File: front/src/common/redux/update-profile/update.profile.api.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import axios from 'axios';
+
+export async function updateProfileApi (data) {
+    console.log(data)
+    
+    const token = localStorage.getItem('token');
+    const user = JSON.parse(localStorage.getItem('user'));
+    console.log(user)
+
+    const studentURL = `${process.env.REACT_APP_UPDATE_STUDENT_PROFILE}${user.id}`;
```
Comment (Alan18081) [inline front/src/common/redux/update-profile/update.profile.api.js:7]: Does this URL actually work?

File: front/src/common/redux/root-saga.js [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 import { studentsSaga } from './students/students.saga';
 import { authSaga } from './auth/auth.saga';
+import { updateSaga } from './update-profile/update.profile.saga';
 import { teachersSaga } from './teachers/teachers.saga';
 import { passwordSaga } from './set-password/set.password.saga';
 
 export default function* startForman() {
   yield fork(studentsSaga);
   yield fork(authSaga);
+  yield fork(updateSaga);
```
Comment (Alan18081) [inline front/src/common/redux/root-saga.js]: Please, rename it to something like `updateUserProfileSaga` to be much more clear

File: front/src/common/redux/update-profile/update.profile.api.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import axios from 'axios';
+
+export async function updateProfileApi (data) {
+    console.log(data)
```
Comment (Alan18081) [inline front/src/common/redux/update-profile/update.profile.api.js]: Useless `console.log`

File: front/src/common/redux/update-profile/update.profile.api.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    const studentURL = `${process.env.REACT_APP_UPDATE_STUDENT_PROFILE}${user.id}`;
+    const teacherURL = `${process.env.REACT_APP_UPDATE_TEACHER_PROFILE}${user.id}`;
+    
+    const headers = { headers: { 
+        'Content-Type': 'application/json',
+        'Authorization': `Bearer ${token}` }
+    };
+
+    if (user.isAdmin) {
+        return axios
```
Comment (Alan18081) [inline front/src/common/redux/update-profile/update.profile.api.js]: Simplify it to the one variable like `const url = user.isAdmin ? teacherURL : studentURL` and write only one actual http call

File: front/src/containers/modal-update-profile/index.js [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+        this.handleCancel();
+        console.log(data);
+        console.log(this.props);
+    
+    }
+
+    render() {
+        const { errorMessage } = this.props;
+        const { user } = this.state;
+console.log(user)
```
Comment (Alan18081) [inline front/src/containers/modal-update-profile/index.js]: Useless `console.log`

File: front/src/containers/modal-update-profile/index.js [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    handleCancel = e => {
+        this.setState({
+            visible: false,
+        });
+    };
+
+    handleUpdateProfile (data) {
+        this.props.updateUserProfileAction(data);
+        this.handleCancel();
+        console.log(data);
```
Comment (Alan18081) [inline front/src/containers/modal-update-profile/index.js]: Useless `console.log`

File: front/src/containers/modal-update-profile/index.js [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+        console.log(this.props);
```
Comment (Alan18081) [inline front/src/containers/modal-update-profile/index.js]: Useless `console.log`

File: front/src/common/redux/update-profile/update.profile.reducer.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import * as types from './types';
+
+const initialState = {
+    isUpdatedProfile: false,
```
Comment (Alan18081) [inline front/src/common/redux/update-profile/update.profile.reducer.js]: Why do you need this boolean field?

File: front/src/components/app/index.js [hunk 9]
```diff
@@ -13,6 +14,7 @@ class App extends Component {
                     <Route path="/student-registration/:token" component={StudentRegistrationPage} />
                     <Route path="/teachers" exact component={TeachersPage} />
                     <Route path="/login" component={LoginPage} />
+                    <Route path="/dashboard" component={UpdateProfile} />
```
Comment (Alan18081) [inline front/src/components/app/index.js:20]: It should be a modal as a part of the header and be available on any page. It should not be a separated route

File: front/src/common/redux/update-profile/update.profile.api.js (hunk 3 above)
Comment (OksanaKozhukh) [reply front/src/common/redux/update-profile/update.profile.api.js]: Outdated

File: front/src/containers/modal-update-profile/index.js (hunk 5 above)
Comment (OksanaKozhukh) [reply front/src/containers/modal-update-profile/index.js]: Outdated

File: front/src/containers/modal-update-profile/index.js (hunk 6 above)
Comment (OksanaKozhukh) [reply front/src/containers/modal-update-profile/index.js]: Deleted

File: front/src/containers/modal-update-profile/index.js (hunk 7 above)
Comment (OksanaKozhukh) [reply front/src/containers/modal-update-profile/index.js]: Deleted

File: front/src/common/redux/root-saga.js (hunk 2 above)
Comment (OksanaKozhukh) [reply front/src/common/redux/root-saga.js]: Renamed

File: front/src/components/app/index.js (hunk 9 above)
Comment (OksanaKozhukh) [reply front/src/components/app/index.js:20]: We don't have header in our app yet, that's why I render this modal on a separated route

File: front/src/common/redux/update-profile/update.profile.api.js (hunk 4 above)
Comment (OksanaKozhukh) [reply front/src/common/redux/update-profile/update.profile.api.js]: Done

File: front/src/common/redux/update-profile/update.profile.api.js (hunk 1 above)
Comment (OksanaKozhukh) [reply front/src/common/redux/update-profile/update.profile.api.js:7]: Yes, it is used to change profile info for students

File: front/src/common/redux/update-profile/update.profile.reducer.js (hunk 8 above)
Comment (OksanaKozhukh) [reply front/src/common/redux/update-profile/update.profile.reducer.js]: Removed

File: front/src/common/redux/auth/auth.reducer.js [hunk 10]
```diff
@@ -4,7 +4,7 @@ import { getUserType } from '../../functions';
 const initialState = {
     isLoggedIn: false,
     errorMessage: '',
-    role: getUserType(),
+    //role: getUserType(),
```
Comment (Alan18081) [inline front/src/common/redux/auth/auth.reducer.js]: Please, remove commented line if you don't need it

Comment (OksanaKozhukh) [reply front/src/common/redux/auth/auth.reducer.js]: Outdated

**Review by Alan18081 — APPROVED** [review]

**Review by Alan18081 — APPROVED** [review]

**Pull request merged**